client/src/page/viewMasterSchedule.jsx

```diff
@@ -0,0 +1,66 @@
+import React from "react";
+import { getSchedules } from "../backend-requests/schoolSchedule";
+import { useState, useEffect} from "react";
+import { DataGrid } from '@mui/x-data-grid';
+
+
+function ViewMasterSchedule(){
+    const [data,setData] = useState(null);
+    const [rows,setRows] = useState([]);
+    const [check,setCheck] = useState(0);
+
+    const columns = [
+    { field: 'id', headerName: 'ID', headerClassName: "header", width: 70 },
+    { field: 'initials', headerName: 'Teacher', headerClassName: "header",width: 100 },
+    { field: 'title', headerName: 'Course Title',headerClassName: "header", width: 250 },
+    { field: 'courseCode', headerName: 'Course Code',headerClassName: "header", width: 130 },
+    { field: 'period', headerName: 'Period',headerClassName: "header", width: 100 },
+    { field: 'location', headerName: 'Location',headerClassName: "header", width: 130 },  
+    ];
+    function getSchedule (){
+        getSchedules().then(response => response.json()).then(dataIn => {setData(dataIn)})
+        if(check === 0){
+            setCheck(1);
+        }
+    }
+
+    function filterSchedule(data){
+        let temprows = [];
+        if(data){
+            for(let i = 0; i < data.length;i++){
+                let schedule = data[i];
+                let clas = schedule.classes;
+                for(let j = 0; j < clas.length;j++){
+                    let title = "";
+                    let courseCode = "";
+                    let c = clas[j].class;
+                    (c) ? (title = c.title) : (title = "");
+                    (c) ? (courseCode = c.courseCode) : (courseCode = "");
+                    temprows.push({"id": clas[j].id,"initials": schedule.teacher.initials, "title": title, "courseCode": courseCode,
+                    "period": clas[j].period, "location": clas[j].location});                
+                }
+            }
+        }
+        return temprows;
+    }
+    
+    useEffect(() => {
+        getSchedule();
+    },[check])
+    useEffect(() => {
+        setRows(filterSchedule(data));
+    },[data])
+    return(
+        <div style={{ "& .header": {backgroundColor: "#a1f0ee"},
```

Comment:
backgroundColor doesn't seem to cover the whole width of the datagrid here.. are you able to put this in its own css file so it's easier to tweak later? Just thinking if we mess with colors or other styling stuff in the last couple days these might be hard to find

Reply:
I tried but I could not get it to work. I will just notify whoever does tweak it to look here.